pvxarray/vtk_source.py

```diff
@@ -189,42 +185,107 @@ def z_index(self, z_index: int):
         self.Modified()
 
     @property
-    def data_range(self):
-        return self.data_array.min(), self.data_array.max()
+    def slicing(self):
+        return self._slicing
 
-    def RequestData(self, request, inInfo, outInfo):
-        # Use open data_array handle to fetch data at
-        # desired Level of Detail
-        try:
-            if self._time is not None:
-                da = self.data_array[{self._time: self.time_index}]
-            else:
-                da = self.data_array
+    @slicing.setter
+    def slicing(self, slicing: int):
```

Comment:
Looks like the `slicing` type hint is wrong, definitely not an `int`.
Off the top of my head I would say it is a `Optional[dict[str, slice]]`

Also, and this is just a suggestion, we could use python's own `slice(start, stop, step)` object to represent slices, instead of an arbitrary 3 element array `[start, stop, step]`

Reply:
Ok, I changed the type to `Optional[List[int]]` in [e25e281](https://github.com/pyvista/pyvista-xarray/pull/58/commits/e25e281f257ccff85b58b7070d95e2258139d17f)